util/db/dbutil.go

```diff
@@ -296,7 +303,12 @@ func (db *Accessor) atomic(fn idemFn, commitLocker sync.Locker, extras ...interf
 			break
 		}
 
-		err = guardedFn(tx)
+		// create a transaction context data
+		txContextData := &txExecutionContext{
+			deadline: atomicDeadline,
```

Comment:
It seems to me that the two variables: atomicDeadline and txContextData.deadline are serving the same purpose.
This is confusing me. 
Why not initialize txContextData at the top, where atomicDeadline is set, and not use the atomicDeadline variable?
This way, atomicDeadline does not need to be updated with txContextData.deadline.

Reply:
The intention here was to avoid a failed transactions from modifying the execution duration warning deadline. 
It's for the last ( or successful ) transaction to update it, as it sees fit.
If the transaction fail, we will have an error message printed out, so that the warning is not an issue anyway.